src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -516,6 +516,12 @@ void MavlinkReceiver::handle_message_command_both(mavlink_message_t *msg, const
 
 		} else if (cmd_mavlink.command == MAV_CMD_LOGGING_STOP) {
 			_mavlink->request_stop_ulog_streaming();
+
+		} else if (cmd_mavlink.command == MAV_CMD_DO_CHANGE_SPEED) {
+			if (_vehicle_status.nav_state == vehicle_status_s::NAVIGATION_STATE_OFFBOARD) {
```

Comment:
Offboard-noob question here: why is MAV_CMD_DO_CHANGE_SPEED the only command for which we need to check the current nav_state first if really in offboard mode before applying it?

Reply:
If you don't check if you're in offboard mode before calling set_offb_cruising_speed it would also change the cruise speed for offboard mode even if you sent a change speed command when in another flight mode. Which could lead to unexpected behaviour if you switch to offboard later on.

Reply:
Okay, and how is it handled for other commands, like position setpoints? I guess the same is true there, you don't want them to have effect unless you're in offboard no?

Reply:
For offboard mode the position setpoints aren't sent as one-time commands, but as a stream of messages. Which are then handled in e.g. https://github.com/PX4/PX4-Autopilot/blob/378b41af6a91cb5dfcabb24421f0a0c9e21a98e5/src/modules/mavlink/mavlink_receiver.cpp#L974. Which in turn checks if offboard is enabled: https://github.com/PX4/PX4-Autopilot/blob/378b41af6a91cb5dfcabb24421f0a0c9e21a98e5/src/modules/mavlink/mavlink_receiver.cpp#L1035. 

Actually, if it makes more sense I could rewrite my code to use the same structure: 
`vehicle_control_mode_s control_mode{};
			_control_mode_sub.copy(&control_mode);

			if (control_mode.flag_control_offboard_enabled) {
                                 set_offb_cruising_speed(cmd_mavlink.param2)
                        }`

Previously with the automatic resets I needed to track the nav_state, but that is no longer true. So it might be more efficient to use _control_mode_sub.copy and get rid of _vehicle_status.

Reply:
>For offboard mode the position setpoints aren't sent as one-time commands, but as a stream of messages. Which are then handled in e.g.

Ah yes that makes sense.

>Previously with the automatic resets I needed to track the nav_state, but that is no longer true. So it might be more efficient to use _control_mode_sub.copy and get rid of _vehicle_status.

That I was also wondering. Would further reduce the complexity and diff to current state.

Reply:
Ok, I removed the _vehicle_status. It's true that the diff looks nicer now :) 

One thing I overlooked previously is that I need to check the vehicle type (FW or MC) in the set and get functions, so I need to use 
```
vehicle_status_s vehicle_status{};
_vehicle_status_sub.copy(&vehicle_status);
```
instead inside these functions. Not sure if it's better for the computational performance in offboard mode, but it definitely doesn't affect any other mode now.

Reply:
Yes, looks nicely contained now.